node/services/sellerService.ts

```diff
@@ -1,74 +1,117 @@
-import { settings } from '../utils'
+import { settings } from './utils'
 
 export default {
   all: async ({ ctx }: { ctx: Context }) => {
-    const sellers = await ctx.clients.sellersClient.sellers()
-    const adyenAccounts = await ctx.clients.account.all()
-    const settingsFetch = await settings(ctx)
-
-    const response = await Promise.all(
-      sellers
-        .reduce((prev, seller) => {
-          if (seller.account === ctx.vtex.account) return prev
-
-          let adyenAccount: SellerAccount | null = null
-
-          for (const account of adyenAccounts) {
-            if (
-              account.sellerId === seller.id &&
-              (!adyenAccount || account.status === 'Active')
-            ) {
-              adyenAccount = account
-              if (account.status === 'Active') {
-                break
+    const {
+      clients: { sellersClient, account: accountClient, adyenClient },
+      vtex: { logger, account: currentAccount },
+    } = ctx
+
+    try {
+      const { items: sellers } = await sellersClient.sellers()
+      const { data: adyenAccounts } = await accountClient.all()
+      const settingsFetch = await settings(ctx)
+
+      const response = await Promise.all(
+        sellers
+          .reduce((prev, seller) => {
+            if (seller.account === currentAccount) return prev
+
+            let adyenAccount: SellerAccount | null = null
+
+            for (const account of adyenAccounts) {
+              if (
+                account.sellerId === seller.id &&
+                adyenAccount?.status !== 'Active' &&
+                (!adyenAccount || account.status === 'Active')
+              ) {
+                adyenAccount = account
```

Comment:
A `break` here would stop further searches

Reply:
Yeah, I was trying to reduce the cognitive complexity score here and thought maybe it'd be ok. To keep the `break`, I pulled out this part out as a separate function.